src/indicator.js

```diff
@@ -306,21 +319,25 @@ const CPUFreqIndicator = new Lang.Class({
 
     _updateFreq: function()
     {
+        let cpufreq = 0, cpucount = 0;
         let lines = Shell.get_file_contents_utf8_sync('/proc/cpuinfo').split('\n');
         for(let i = 0; i < lines.length; i++)
         {
             let line = lines[i];
 
             if(line.search(/cpu mhz/i) < 0)
                 continue;
-            this.cpufreq = parseInt(line.substring(line.indexOf(':') + 2));
-            this.imCurrentLabel.set_text(this._getCurFreq());
-            if(this.lblActive)
-                this.lbl.set_text(this._getCurFreq());
-            else
-                this.lbl.set_text('');
-            break;
+
+			let f = Shell.get_file_contents_utf8_sync('/sys/devices/system/cpu/cpu' + cpucount++ + '/cpufreq/scaling_cur_freq');
+			cpufreq += parseInt(f / 1024);
         }
+		this.cpufreq = (cpufreq / cpucount)
+		this.imCurrentLabel.set_text(this._getCurFreq());
+		if(this.lblActive)
+			this.lbl.set_text(this._getCurFreq());
+		else
+			this.lbl.set_text('');
+
```

Comment:
Again misleading indentation, but the calculation looks good

Reply:
Just fixed all the indentations -- my vimrc was the issue, I implemented a new formatter/indenter.